homeassistant/components/cover/group.py

```diff
@@ -0,0 +1,268 @@
+        async_track_state_change(self.hass, self._entities,
+                                 state_change_listener)
+
+    @property
+    def name(self):
+        """Return the name of the cover."""
+        return self._name
+
+    @property
+    def assumed_state(self):
+        """Enable buttons even if at end position."""
+        return True
+
+    @property
+    def should_poll(self):
+        """Disable polling for the MultiCover."""
+        return False
+
+    @property
+    def supported_features(self):
+        """Flag supported features for the cover."""
+        return self._supported_features
+
+    @property
+    def is_closed(self):
+        """Return if all covers in group are closed."""
+        return self._is_closed
+
+    @property
+    def current_cover_position(self):
+        """Return current position for all covers."""
+        return self._cover_position
+
+    @property
+    def current_cover_tilt_position(self):
+        """Return current tilt position for all covers."""
+        return self._tilt_position
+
+    async def async_open_cover(self, **kwargs):
+        """Move the covers up."""
+        self.hass.async_add_job(open_cover, self.hass,
+                                self._covers[KEY_OPEN_CLOSE])
+
+    async def async_close_cover(self, **kwargs):
+        """Move the covers down."""
+        self.hass.async_add_job(close_cover, self.hass,
+                                self._covers[KEY_OPEN_CLOSE])
+
+    async def async_stop_cover(self, **kwargs):
+        """Fire the stop action."""
+        self.hass.async_add_job(stop_cover, self.hass,
+                                self._covers[KEY_STOP])
+
+    async def async_set_cover_position(self, **kwargs):
+        """Set covers position."""
+        position = kwargs[ATTR_POSITION]
+        self.hass.async_add_job(set_cover_position, self.hass,
+                                position, self._covers[KEY_POSITION])
+
+    async def async_open_cover_tilt(self, **kwargs):
+        """Tilt covers open."""
+        self.hass.async_add_job(open_cover_tilt, self.hass,
+                                self._tilts[KEY_OPEN_CLOSE])
+
+    async def async_close_cover_tilt(self, **kwargs):
+        """Tilt covers closed."""
+        self.hass.async_add_job(close_cover_tilt, self.hass,
+                                self._tilts[KEY_OPEN_CLOSE])
+
+    async def async_stop_cover_tilt(self, **kwargs):
+        """Stop cover tilt."""
+        self.hass.async_add_job(stop_cover_tilt, self.hass,
+                                self._tilts[KEY_STOP])
+
+    async def async_set_cover_tilt_position(self, **kwargs):
+        """Set tilt position."""
+        tilt_position = kwargs[ATTR_TILT_POSITION]
+        self.hass.async_add_job(set_cover_tilt_position, self.hass,
+                                tilt_position, self._tilts[KEY_POSITION])
+
+    async def async_update(self):
+        """Update state and attributes."""
+        def update_is_closed():
```

Comment:
Don't define these functions inside `async_update`. Make them regular methods on the entity.